### ISSUE TITLE
Bug in algorithm for weighted nodes

### ISSUE THREAD

**Issue by grihabor** [serialx/hashring#22]
I am trying to iterate over all nodes of the hash ring, and there are missing nodes.

Here is a sample:
```
package main

import (
	"github.com/serialx/hashring"
	"fmt"
	"sort"
)

func generate(n int) map[string]int {
	result := make(map[string]int)
	for i := 0; i < n; i++ {
		result[fmt.Sprintf("node%02d", i)] = i
	}
	return result
}

func main() {
	weights := generate(1000)
	ring := hashring.NewWithWeights(weights)
	fmt.Printf("size %d\n", ring.Size())

	var nodes []string
	var ok bool
	for i := 0; i < ring.Size(); i++ {
		nodes, ok = ring.GetNodes("1", ring.Size()-i)
		if ok {
			break
		}
	}
	sort.Strings(nodes)
	fmt.Printf("actual %d\n", len(nodes))
}
```
Result:
```
size 1000
actual 987
```

**Comment by grihabor** [comment]
I think the problem is around here https://github.com/serialx/hashring/blob/7706f26af1941a4e865d509c4681eb86c3b62bf8/hashring.go#L104

**Comment by serialx** [comment]
Fixed by #23
